XCast/XCast.cpp

```diff
@@ -373,7 +373,10 @@ void XCast::onLocateCastTimer()
     int status = _rtConnector->connectToRemoteService();
     if(status != 0)
     {
-        locateCastObjectRetryCount++;
+        if(locateCastObjectRetryCount < 4)
```

Comment:
instead of making this change just update 

        if(locateCastObjectRetryCount == 4)
condition to 
        if(locateCastObjectRetryCount >= 4)
 
so in case if we print locateCastObjectRetryCount we know after how many attempt it got connection success

Reply:
This approach might introduce an undefined behavior in a (very unlikely) scenario of integer overflow. Is it worth the possibility to track how many connection attempts were done?

Reply:
@apatel859 could you answer whether you agree with such risk of integer overflow?

Reply:
@dawid-trendota-red 
I would lean towards this conditions (mentioned by @apatel859 ):
if(locateCastObjectRetryCount >= 4)

If you believe overflowing locateCastObjectRetryCount is a real problem (rather unlikely, as most modern platforms use 32bit int), please change the declarion of locateCastObjectRetryCount to uint64_t.